Sources/TuistKit/ProjectEditor/ProjectEditor.swift

```diff
@@ -0,0 +1,56 @@
+import Basic
+import Foundation
+import TuistGenerator
+import TuistSupport
+
+protocol ProjectEditing: AnyObject {
+    /// Generates an Xcode project to edit the Project defined in the given directory.
+    /// - Parameters:
+    ///   - at: Directory whose project will be edited.
+    ///   - destinationDirectory: Directory in which the Xcode project will be generated.
+    func edit(at: AbsolutePath, in destinationDirectory: AbsolutePath) throws
+}
+
+final class ProjectEditor: ProjectEditing {
+    /// Project generator.
+    let generator: Generating
+
+    /// Project editor mapper.
+    let projectEditorMapper: ProjectEditorMapping
+
+    /// Utility to locate Tuist's resources.
+    let resourceLocator: ResourceLocating
```

Comment:
A note for a follow-up PR: We have several file/folder locators that I think we can merge into a single locator that has multiple methods, each for every resource being located.